### ISSUE TITLE
line-height does not render in the middle

### ISSUE THREAD

**Issue by jorgenhorstink** [dompdf/dompdf#670]
Hi,

Yesterday I've been digging into the source code, but I was not able to isolate the issue. Line-height is not rendered in the middle as a webbrowser does;

```
<!DOCTYPE html>
<html>
    <body>
        <div style="font-size: 40px; line-height: 400px; border: 1px solid red; width: 500px;">Not in the middle</div>
    </body>
</html>
```

I cloned the most recent version of the source code, and I am still able to reproduce the issue. I do not know the details about how line-height should be calculated and rendered, but I think it is a bug. Am I right, if so, any ideas how to fix the issue? I think it has something to do with the way the position of line boxes is calculated.

_expected result_
The text Not in the middle, should be in the middle.

**Comment by Paroca72** [comment]
This bug is still open on the 0.8.3 version!

**Comment by bsweeney** [comment]
Indeed. Dompdf currently handles line-height mostly at the block level. Using the example provided earlier you can get a better rendering if you style the span elements with `display: inline-block;`.

**Comment by bsweeney** [comment]
alignment is generally improved with #2726 but going through the samples here I do still see some room for improvment

**Comment by yorunairu** [comment]
![image](https://github.com/user-attachments/assets/a48f1e97-18eb-450c-b40f-e952438cc979)
```
<td class="text-left v-align-middle" style="line-height: 25%">
    {!! $item->description !!}
</td>
```

having issue render value from summernote wysiwyg, line height doesnt work properly.

**Comment by bsweeney** [comment]
@yorunairu hard to tell if this is related or not without a full example of the code. Testing with what you've provided does not render the same. Also, which version of Dompdf are you using?
